src/Avalonia.Base/Input/KeyboardDevice.cs

```diff
@@ -2,163 +2,28 @@
 using System.Runtime.CompilerServices;
 using Avalonia.Input.Raw;
 using Avalonia.Input.TextInput;
-using Avalonia.Interactivity;
 using Avalonia.VisualTree;
 
 namespace Avalonia.Input
 {
     public class KeyboardDevice : IKeyboardDevice, INotifyPropertyChanged
     {
         private IInputElement? _focusedElement;
-        private IInputRoot? _focusedRoot;
+        private readonly TextInputMethodManager _textInputManager = new TextInputMethodManager();
 
         public event PropertyChangedEventHandler? PropertyChanged;
 
         public static IKeyboardDevice? Instance => AvaloniaLocator.Current.GetService<IKeyboardDevice>();
 
         public IInputManager? InputManager => AvaloniaLocator.Current.GetService<IInputManager>();
 
-        public IFocusManager? FocusManager => AvaloniaLocator.Current.GetService<IFocusManager>();
-        
-        // This should live in the FocusManager, but with the current outdated architecture
-        // the source of truth about the input focus is in KeyboardDevice
-        private readonly TextInputMethodManager _textInputManager = new TextInputMethodManager();
-
         public IInputElement? FocusedElement => _focusedElement;
```

Comment:
I suppose it can be marked as obsolete. 
In the future I want to see current Device API removed or refactored (not in 11.0 most likely, maybe 12.0). Good to see it doesn't handle focus anymore in this PR. See: https://github.com/AvaloniaUI/Avalonia/issues/7939